lib/_stream_readable.js

```diff
@@ -447,7 +447,8 @@ function maybeReadMore_(stream, state) {
 // for virtual (non-string, non-buffer) streams, "length" is somewhat
 // arbitrary, and perhaps not very meaningful.
 Readable.prototype._read = function(n) {
-  this.emit('error', new Error('not implemented'));
+  const err = new Error('not implemented');
+  process.nextTick(() => this.emit('error', err));
 };
```

Comment:
oh yea sorry about that. that probably works. also running the benchmarks will probably show if this is still a problem in general. @mcollina is more of an expert on this than I am